include/benchmark/benchmark.h

```diff
@@ -746,13 +746,13 @@ class BENCHMARK_EXPORT State {
   // have been called previously.
   //
   // NOTE: KeepRunning may not be used after calling either of these functions.
-  BENCHMARK_ALWAYS_INLINE StateIterator begin();
-  BENCHMARK_ALWAYS_INLINE StateIterator end();
+  inline BENCHMARK_ALWAYS_INLINE StateIterator begin();
```

Comment:
should we just have a definition for `BENCHMARK_ALWAYS_INLINE` for WINDOWS && !CLANG or something?

Reply:
The issue is that the *definition* was marked as `inline`, while *declaration* was not.
In windows, that effectively results in two different function declarations.
So i don't think this is `BENCHMARK_ALWAYS_INLINE`-specific in any way,
and thus i don't see why `BENCHMARK_ALWAYS_INLINE` should be altered.

Reply:
it reads a bit surprisingly to have an "always inline" attribute but then have to supplement it with "inline".  i assumed you needed this because on some platforms "always inline" defines to "".

Reply:
As far as i understand it, `BENCHMARK_ALWAYS_INLINE` is an optimization hint
that does not contain the same binding semantical meaning (on the symbol visibility)
as the  `inline` one.